### PR TITLE
[MU4] Play note on click

### PR DESCRIPTION
**Pull request by igorkorsukov**
This is a solution with independent use synthesizer, this is cool, but not effective, so will be redesigned.

### CODE REVIEW

File: mu4/scenes/playback/internal/playbackcontroller.h [hunk 1]
```diff
@@ -50,6 +52,8 @@ class PlaybackController : public IPlaybackController, public actions::Actionabl
     float playbackPosition() const override;
     async::Channel<uint32_t> midiTickPlayed() const override;
 
+    void playElementOnClick(const domain::notation::Element* e) override;
```
Comment (Eism) [inline mu4/scenes/playback/internal/playbackcontroller.h:55]: Let's just call it playElement?

File: mu4/scenes/playback/internal/playbackconfiguration.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#include "playbackconfiguration.h"
+
+using namespace mu::scene::playback;
+
+bool PlaybackConfiguration::isPlayElementOnClick() const
```
Comment (Eism) [inline mu4/scenes/playback/internal/playbackconfiguration.cpp:23]: Why do you save the fact of playing an element on click in the settings? Or does it return the ability to play the element on click?

File: mu4/scenes/playback/internal/playbackcontroller.h (hunk 1 above)
Comment (igorkorsukov) [reply mu4/scenes/playback/internal/playbackcontroller.h:55]: At first I called it that.
But there is a setting to play elements on click or not.
So I decided to call this method `playElementOnClick`
And in the future, if necessary, add a simple `playElement` method, which should not depend on any settings, but should certainly play the element.

File: mu4/scenes/playback/internal/playbackconfiguration.cpp (hunk 2 above)
Comment (igorkorsukov) [reply mu4/scenes/playback/internal/playbackconfiguration.cpp:23]: second
This is the setting - whether the element should be played on click

File: framework/audio/midi/miditypes.h [hunk 3]
```diff
@@ -215,6 +215,7 @@ struct MidiData {
 struct MidiStream {
     MidiData initData;
 
+    bool isAllowStreaing = false;
```
Comment (Eism) [inline framework/audio/midi/miditypes.h:218]: isAllowStreaming

Comment (igorkorsukov) [reply framework/audio/midi/miditypes.h]: fixed

File: framework/audio/engine/internal/audioplayer.cpp [hunk 4]
```diff
@@ -280,3 +280,30 @@ void AudioPlayer::onMidiStatusChanged(engine::IAudioEngine::Status status)
         onStop();
     }
 }
+
+void AudioPlayer::playMidi(const midi::MidiData& data)
+{
+    if (!m_singleNoteMidiSource) {
```
Comment (vpereverzev) [inline framework/audio/engine/internal/audioplayer.cpp:286]: I think it'd be better to not mention anything related with "notes" here

Comment (igorkorsukov) [reply framework/audio/engine/internal/audioplayer.cpp:286]: I agree, in one PR this will be fixed

**Pull request merged**